lib/iris/analysis/__init__.py

```diff
@@ -1584,27 +1584,19 @@ def _lazy_max_run(array, axis=-1, **kwargs):
 
 
 def _rms(array, axis, **kwargs):
-    # XXX due to the current limitations in `da.average` (see below), maintain
-    # an explicit non-lazy aggregation function for now.
-    # Note: retaining this function also means that if weights are passed to
-    # the lazy aggregator, the aggregation will fall back to using this
-    # non-lazy aggregator.
-    rval = np.sqrt(ma.average(np.square(array), axis=axis, **kwargs))
-    if not ma.isMaskedArray(array):
-        rval = np.asarray(rval)
```

Comment:
I've not preserved this handling to revert to a plain numpy array because it's inconsistent with the MEAN aggregator, and I don't think there is any way to make the lazy case equivalent.  Note that `aggregated_by` has its own handling anyway:

https://github.com/SciTools/iris/blob/d1f3e4564e09de1e269872f5d14d4d3bf3ce5489/lib/iris/cube.py#L4167-L4174

though I can't see any equivalent of that in `collapsed` or `rolling_window`....